### DISCUSSION TITLE
Compatibility issue between axum 0.7.5 and tower-http 0.6.0

### DISCUSSION THREAD

**Discussion by kanpov** [tokio-rs/axum#2926] [Q&A] [answered]
### Summary

Tower 0.5.x is out and tower-http 0.6.0 upgraded to it, but axum 0.7.x has not and I don't really know how long it'll take until 0.8.x is actually out with tower 0.5.x compatibility.

Hence, tower-http 0.6.x and axum 0.7.x are incompatible. Is there any feasible solution except for downgrading tower-http to 0.5.x?

### axum version

0.7.5

**Comment by jplatte** [comment, marked as answer]
`tower 0.5` still depends on `tower-service 0.3` and `tower-layer 0.3`, same as `tower 0.4`. All the packages you mentioned are compatible with each other, with only a tiny exception of `tower-http`'s `ServiceBuilderExt` only working with the `ServiceBuilder` of the right `tower` version.

Also, `axum 0.7.6` (released yesterday) already did the upgrade to `tower-http 0.6` (for its internal needs).

**Reply by kanpov** [reply]
> `tower 0.5` still depends on `tower-service 0.3` and `tower-layer 0.3`, same as `tower 0.4`. All the packages you mentioned are compatible with each other, with only a tiny exception of `tower-http`'s `ServiceBuilderExt` only working with the `ServiceBuilder` of the right `tower` version.
> 
> Also, `axum 0.7.6` (released yesterday) already did the upgrade to `tower-http 0.6` (for its internal needs).

Thanks for the clarification!
